From b1931eebbecc91a0cddfedd518cceb0a0897f507 Mon Sep 17 00:00:00 2001
From: Jeroen Hofstee <jhofstee@victronenergy.com>
Date: Thu, 4 Aug 2016 10:34:14 +0200
Subject: [PATCH] add dummy distclean

---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index 88ff456..5e94cd4 100644
--- a/Makefile
+++ b/Makefile
@@ -30,6 +30,7 @@ help :
 	@ echo " help - print this message"
 	@ echo " install - install everything"
 
+distclean: ;
 clean: ;
 
 install_app : $(FILES)
-- 
1.9.1

