From ff09859487c3adacc5da151c9ebb9a74edd8d16d Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Fri, 14 Sep 2018 16:04:01 +0100
Subject: [PATCH] mutex: always use unlock_slowpath

---
 kernel/locking/mutex.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/kernel/locking/mutex.c b/kernel/locking/mutex.c
index 858a07590e39..2a1963b93274 100644
--- a/kernel/locking/mutex.c
+++ b/kernel/locking/mutex.c
@@ -607,10 +607,6 @@ static noinline void __sched __mutex_unlock_slowpath(struct mutex *lock, unsigne
  */
 void __sched mutex_unlock(struct mutex *lock)
 {
-#ifndef CONFIG_DEBUG_LOCK_ALLOC
-	if (__mutex_unlock_fast(lock))
-		return;
-#endif
 	__mutex_unlock_slowpath(lock, _RET_IP_);
 }
 EXPORT_SYMBOL(mutex_unlock);
-- 
2.19.0

