From d19a289de074130aa73aace7ce54caec49d8fd84 Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Fri, 8 Feb 2019 14:05:56 +0000
Subject: [PATCH 1/2] Don't spam syslog on reload and not using hosts files

---
 src/cache.c | 13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/src/cache.c b/src/cache.c
index 2f2c519cd49a..d578ff90bcc1 100644
--- a/src/cache.c
+++ b/src/cache.c
@@ -1290,16 +1290,11 @@ void cache_reload(void)
 	  }
       }
 	
-  if (option_bool(OPT_NO_HOSTS) && !daemon->addn_hosts)
-    {
-      if (daemon->cachesize > 0)
-	my_syslog(LOG_INFO, _("cleared cache"));
-    }
-  else
-    {
-      if (!option_bool(OPT_NO_HOSTS))
-	total_size = read_hostsfile(HOSTSFILE, SRC_HOSTS, total_size, (struct crec **)daemon->packet, revhashsz);
+  if (!option_bool(OPT_NO_HOSTS))
+    total_size = read_hostsfile(HOSTSFILE, SRC_HOSTS, total_size, (struct crec **)daemon->packet, revhashsz);
       
+  if (daemon->addn_hosts)
+    {
       daemon->addn_hosts = expand_filelist(daemon->addn_hosts);
       for (ah = daemon->addn_hosts; ah; ah = ah->next)
 	if (!(ah->flags & AH_INACTIVE))
-- 
2.30.0

