From db039148ac544c9fa0792c92470f2c68a6b375d6 Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Fri, 8 Feb 2019 15:11:32 +0000
Subject: [PATCH 2/2] Log only changes to nameserver list

---
 src/dnsmasq.h |  1 +
 src/network.c | 21 ++++++++++++++++++---
 2 files changed, 19 insertions(+), 3 deletions(-)

diff --git a/src/dnsmasq.h b/src/dnsmasq.h
index f53e9a5be6cd..0efb1e5a456a 100644
--- a/src/dnsmasq.h
+++ b/src/dnsmasq.h
@@ -524,6 +524,7 @@ union mysockaddr {
 #define SERV_LOOP           8192  /* server causes forwarding loop */
 #define SERV_DO_DNSSEC     16384  /* Validate DNSSEC when using this server */
 #define SERV_GOT_TCP       32768  /* Got some data from the TCP connection */
+#define SERV_NEW           65536  /* server was just added */
 
 struct serverfd {
   int fd;
diff --git a/src/network.c b/src/network.c
index 8ae7a7061dc8..1881720f685e 100644
--- a/src/network.c
+++ b/src/network.c
@@ -1345,6 +1345,12 @@ void pre_allocate_sfds(void)
       }  
 }
 
+static void log_gone(struct server *serv)
+{
+  int port = prettyprint_addr(&serv->addr, daemon->namebuff);
+  my_syslog(LOG_INFO, _("stopped using nameserver %s#%d"), daemon->namebuff, port);
+}
+
 void mark_servers(int flag)
 {
   struct server *serv;
@@ -1372,6 +1378,7 @@ void cleanup_servers(void)
       if (serv->flags & SERV_MARK)
        {
          server_gone(serv);
+         log_gone(serv);
          *up = serv->next;
          if (serv->domain)
 	   free(serv->domain);
@@ -1410,8 +1417,9 @@ void add_update_server(int flags,
 	    if (serv->flags & SERV_HAS_DOMAIN)
 	      continue;
 	  }
-	
-        break;
+
+	if (!memcmp(addr, &serv->addr, sizeof(*addr)))
+	  break;
       }
 
   if (serv)
@@ -1441,6 +1449,8 @@ void add_update_server(int flags,
 	  if (domain)
 	    strcpy(domain_str, domain);
 	}
+
+      flags |= SERV_NEW;
     }
   
   if (serv)
@@ -1552,7 +1562,12 @@ void check_servers(void)
 	  if (serv->sfd)
 	    serv->sfd->used = 1;
 	}
-      
+
+      if (!(serv->flags & SERV_NEW))
+	continue;
+
+      serv->flags &= ~SERV_NEW;
+
       if (!(serv->flags & SERV_NO_REBIND) && !(serv->flags & SERV_LITERAL_ADDRESS))
 	{
 	  if (++count > SERVERS_LOGGED)
-- 
2.26.2

