From f2a11bbaccd4ce706db077c1849413dd3bed63b9 Mon Sep 17 00:00:00 2001
From: Jeroen Hofstee <jhofstee@victronenergy.com>
Date: Tue, 20 Jul 2021 17:57:59 +0200
Subject: [PATCH] do not copy /etc/localtime

---
 initscript/debian/avahi-daemon.in | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/initscript/debian/avahi-daemon.in b/initscript/debian/avahi-daemon.in
index 49ec358..a3d705a 100755
--- a/initscript/debian/avahi-daemon.in
+++ b/initscript/debian/avahi-daemon.in
@@ -124,14 +124,6 @@ d_start() {
     modprobe capability >/dev/null 2>&1 || true
 
     $DAEMON -c && return 0
-
-    if [ -s /etc/localtime ]; then
-	if [ ! -d /etc/avahi/etc ]; then
-	    mkdir -p @sysconfdir@/avahi/etc >/dev/null 2>&1
-	fi
-	cp -fp /etc/localtime @sysconfdir@/avahi/etc >/dev/null 2>&1
-    fi;
-    
     $DAEMON -D
 }
 
-- 
2.17.1

