From 298971b3e69205f581680793d9bba3c9de5171e6 Mon Sep 17 00:00:00 2001
From: Jeroen Hofstee <jhofstee@victronenergy.com>
Date: Wed, 20 Aug 2014 17:55:27 +0200
Subject: [PATCH 2/2] svscanboot: update paths

---
 src/svscanboot.sh | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/svscanboot.sh b/src/svscanboot.sh
index 74079ea..d78df0b 100644
--- a/src/svscanboot.sh
+++ b/src/svscanboot.sh
@@ -1,11 +1,10 @@
-
-PATH=/command:/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin
+PATH=/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin
 
 exec </dev/null
 exec >/dev/null
 exec 2>/dev/null
 
-/command/svc -dx /service/* /service/*/log
+/bin/svc -dx /service/* /service/*/log
 
 env - PATH=$PATH svscan /service 2>&1 | \
 env - PATH=$PATH readproctitle service errors: ................................................................................................................................................................................................................................................................................................................................................................................................................
-- 
1.9.1

