From 0b109d6c4486c9861696b91a41f9ecbf5a2dfef8 Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Fri, 23 May 2025 12:17:52 +0100
Subject: [PATCH] remove bogus expected response length for ModbusUdpClient

---
 pymodbus/transaction.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/pymodbus/transaction.py b/pymodbus/transaction.py
index 949459cb0456..183c26142a1d 100644
--- a/pymodbus/transaction.py
+++ b/pymodbus/transaction.py
@@ -172,8 +172,6 @@ class ModbusTransactionManager(object):
                     c_str = str(self.client)
                     if "modbusudpclient" in c_str.lower().strip():
                         full = True
-                        if not expected_response_length:
-                            expected_response_length = Defaults.ReadSize
                     response, last_exception = self._transact(
                         request,
                         expected_response_length,
-- 
2.49.0

