From d3331624f61e1383e6bdececb09ff69c36acd4b3 Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Fri, 23 May 2025 12:17:52 +0100
Subject: [PATCH] reject responses that fail validation

---
 pymodbus/transaction.py | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/pymodbus/transaction.py b/pymodbus/transaction.py
index 183c26142a1d..a38e13656085 100644
--- a/pymodbus/transaction.py
+++ b/pymodbus/transaction.py
@@ -178,6 +178,7 @@ class ModbusTransactionManager(object):
                         full=full,
                         broadcast=broadcast
                     )
+                    valid_response = False
                     while retries > 0:
                         valid_response = self._validate_response(
                             request, response, expected_response_length
@@ -204,12 +205,15 @@ class ModbusTransactionManager(object):
                                 else:
                                     break
                             # full = False
-                    addTransaction = partial(self.addTransaction,
-                                             tid=request.transaction_id)
-                    self.client.framer.processIncomingPacket(response,
-                                                             addTransaction,
-                                                             request.unit_id)
-                    response = self.getTransaction(request.transaction_id)
+                    if valid_response:
+                        addTransaction = partial(self.addTransaction,
+                                                 tid=request.transaction_id)
+                        self.client.framer.processIncomingPacket(response,
+                                                                 addTransaction,
+                                                                 request.unit_id)
+                        response = self.getTransaction(request.transaction_id)
+                    else:
+                        response = None
                     if not response:
                         if len(self.transactions):
                             response = self.getTransaction(tid=0)
-- 
2.49.0

