From 87c78e2b6df6ec513c558c7ed9f9ec4b92252a75 Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Thu, 10 Jan 2019 14:53:45 +0000
Subject: [PATCH] clock: set hw clock when time is updated over dbus

---
 src/clock.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/clock.c b/src/clock.c
index 0fde2c341aca..9418c6d0d5a0 100644
--- a/src/clock.c
+++ b/src/clock.c
@@ -258,6 +258,8 @@ static DBusMessage *set_property(DBusConnection *conn,
 		if (settimeofday(&tv, NULL) < 0)
 			return __connman_error_invalid_arguments(msg);
 
+		system("hwclock --utc --systohc");
+
 		connman_dbus_property_changed_basic(CONNMAN_MANAGER_PATH,
 				CONNMAN_CLOCK_INTERFACE, "Time",
 				DBUS_TYPE_UINT64, &newval);
-- 
2.20.1

