From 36facbdba32595afb8cbd38785f4b785bb02e6e3 Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Tue, 28 Jan 2020 15:41:26 +0000
Subject: [PATCH 14/16] wifi: ignore connection errors

---
 plugins/wifi.c | 17 +----------------
 1 file changed, 1 insertion(+), 16 deletions(-)

diff --git a/plugins/wifi.c b/plugins/wifi.c
index 521ef9f6c4c4..9a4b9c489051 100644
--- a/plugins/wifi.c
+++ b/plugins/wifi.c
@@ -1990,14 +1990,6 @@ static void connect_callback(int result, GSupplicantInterface *interface,
 
 	DBG("network %p result %d", network, result);
 
-	if (result == -ENOKEY) {
-		connman_network_set_error(network,
-					CONNMAN_NETWORK_ERROR_INVALID_KEY);
-	} else if (result < 0) {
-		connman_network_set_error(network,
-					CONNMAN_NETWORK_ERROR_CONFIGURE_FAIL);
-	}
-
 	connman_network_unref(network);
 }
 
@@ -2314,14 +2306,7 @@ static bool handle_wps_completion(GSupplicantInterface *interface,
 static bool handle_assoc_status_code(GSupplicantInterface *interface,
                                      struct wifi_data *wifi)
 {
-	if (wifi->state == G_SUPPLICANT_STATE_ASSOCIATING &&
-			wifi->assoc_code == ASSOC_STATUS_NO_CLIENT &&
-			wifi->load_shaping_retries < LOAD_SHAPING_MAX_RETRIES) {
-		wifi->load_shaping_retries ++;
-		return TRUE;
-	}
-	wifi->load_shaping_retries = 0;
-	return FALSE;
+	return TRUE;
 }
 
 static bool handle_4way_handshake_failure(GSupplicantInterface *interface,
-- 
2.25.0

