From 8ada323cf51aab89518d6704cab59f8a52137767 Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Tue, 28 Jan 2020 17:21:01 +0000
Subject: [PATCH 15/16] service: do not mark services failed on connect timeout

---
 src/service.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/src/service.c b/src/service.c
index d30a413ccdfc..c2062ae0c89a 100644
--- a/src/service.c
+++ b/src/service.c
@@ -4037,13 +4037,6 @@ static gboolean connect_timeout(gpointer user_data)
 	} else
 		autoconnect = true;
 
-	__connman_service_ipconfig_indicate_state(service,
-					CONNMAN_SERVICE_STATE_FAILURE,
-					CONNMAN_IPCONFIG_TYPE_IPV4);
-	__connman_service_ipconfig_indicate_state(service,
-					CONNMAN_SERVICE_STATE_FAILURE,
-					CONNMAN_IPCONFIG_TYPE_IPV6);
-
 	if (autoconnect &&
 			service->connect_reason !=
 				CONNMAN_SERVICE_CONNECT_REASON_USER)
-- 
2.25.0

