From eec0818f4a8d8a73449dd773ff9772d7edd83bcf Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Tue, 28 Jan 2020 18:26:06 +0000
Subject: [PATCH 16/16] network: do not mark network failed on error while
 associating

---
 src/network.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/network.c b/src/network.c
index 2e423bc362ec..a53b61176c74 100644
--- a/src/network.c
+++ b/src/network.c
@@ -1323,8 +1323,7 @@ int connman_network_set_connected(struct connman_network *network,
 		network, network->connected, connected, network->connecting,
 		network->associating);
 
-	if ((network->connecting || network->associating) &&
-							!connected) {
+	if (network->connecting && !connected) {
 		connman_network_set_error(network,
 					CONNMAN_NETWORK_ERROR_CONNECT_FAIL);
 		return 0;
-- 
2.25.0

