From 09d5d1a55296fc8e1dce760eca3292a7613d854f Mon Sep 17 00:00:00 2001
From: Jeroen Hofstee <jhofstee@victronenergy.com>
Date: Thu, 23 Nov 2017 21:36:12 +0100
Subject: [PATCH] fix makefile

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 43599a9..51599bf 100644
--- a/Makefile
+++ b/Makefile
@@ -18,7 +18,7 @@ DELEGATES = \
 	$(SOURCEDIR)/delegates/servicemapper.py \
 	$(SOURCEDIR)/delegates/vebussocwriter.py \
 	$(SOURCEDIR)/delegates/systemstate.py \
-	$(SOURCEDIR)/delegates/voltagesense.py
+	$(SOURCEDIR)/delegates/voltagesense.py \
 	$(SOURCEDIR)/delegates/batterylife.py
 
 VEDLIB_FILES = \
-- 
2.7.4

