From e613a8f4683b70e7b2562d98065684512ee068b6 Mon Sep 17 00:00:00 2001
From: Jeroen Hofstee <jhofstee@victronenergy.com>
Date: Thu, 9 Jan 2025 14:36:21 +0100
Subject: [PATCH] flush traces

Upstream-Status: Inappropriate [Venus specific]
---
 c/websocket.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/c/websocket.h b/c/websocket.h
index 45acc21..3340164 100644
--- a/c/websocket.h
+++ b/c/websocket.h
@@ -83,6 +83,7 @@ ssize_t ws_send(ws_ctx_t *ctx, const void *buf, size_t len);
     if (! settings.daemon) { \
         fprintf(stream, "  %d: ", settings.handler_id); \
         fprintf(stream, __VA_ARGS__); \
+        fflush(stream); \
     }
 
 #define handler_msg(...) gen_handler_msg(stdout, __VA_ARGS__);
-- 
2.43.0

