From 714f293df642af852838cf2f7d5dc10b56451ed2 Mon Sep 17 00:00:00 2001
From: Jeroen Hofstee <jhofstee@victronenergy.com>
Date: Tue, 2 Feb 2016 12:09:42 +0100
Subject: [PATCH] hardcode kernel version since the command to get it fails

---
 Makefile | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index ea3982b..09597e0 100644
--- a/Makefile
+++ b/Makefile
@@ -84,8 +84,7 @@ mrproper:
 				echo ""							;\
 			done								\
 		) > Kconfig.kernel							;\
-		kver=$$($(MAKE) --no-print-directory -C $(KLIB_BUILD) kernelversion |	\
-			sed 's/^\(\(3\|2\.6\)\.[0-9]\+\).*/\1/;t;d')			;\
+		kver=3.7.1			;\
 		test "$$kver" != "" || echo "Kernel version parse failed!"		;\
 		test "$$kver" != ""							;\
 		kvers="$$(seq 14 39 | sed 's/^/2.6./')"					;\
-- 
1.9.1

