From 34708d84b1b2d922e177545555cb040b50a429d4 Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Tue, 23 Feb 2021 13:22:18 +0000
Subject: [PATCH] rtl8723du: fix build for pre-4.15 kernels

---
 core/rtw_sta_mgt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/rtw_sta_mgt.c b/core/rtw_sta_mgt.c
index b68b55814f64..92d7f52a1956 100644
--- a/core/rtw_sta_mgt.c
+++ b/core/rtw_sta_mgt.c
@@ -410,7 +410,7 @@ static void rtw_init_recv_timer(struct recv_reorder_ctrl *preorder_ctrl)
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 15, 0)
 	timer_setup(&preorder_ctrl->reordering_ctrl_timer, rtw_reordering_ctrl_timeout_handler, 0);
 #else
-	rtw_init_timer(&(preorder_ctrl->reordering_ctrl_timer), adapt, rtw_reordering_ctrl_timeout_handler, preorder_ctrl);
+	rtw_init_timer(&(preorder_ctrl->reordering_ctrl_timer), preorder_ctrl->adapt, rtw_reordering_ctrl_timeout_handler, preorder_ctrl);
 #endif
 }
 
-- 
2.30.1

