From 77a625b6b902a9ea85e5708868a29045bd906c16 Mon Sep 17 00:00:00 2001
From: Jeroen Hofstee <jhofstee@victronenergy.com>
Date: Mon, 26 Jan 2026 14:38:42 +0100
Subject: [PATCH] increase the default dbus timeout to 2 minutes

Upstream-Status: Inappropriate [Venus specific]

---
 dbus/dbus-connection-internal.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dbus/dbus-connection-internal.h b/dbus/dbus-connection-internal.h
index df3c701..c6b9ceb 100644
--- a/dbus/dbus-connection-internal.h
+++ b/dbus/dbus-connection-internal.h
@@ -42,8 +42,8 @@ typedef enum
   DBUS_ITERATION_BLOCK      = 1 << 2  /**< Block if nothing to do. */
 } DBusIterationFlags;
 
-/** default timeout value when waiting for a message reply, 25 seconds */
-#define _DBUS_DEFAULT_TIMEOUT_VALUE (25 * 1000)
+/** default timeout value when waiting for a message reply, 2 minutes */
+#define _DBUS_DEFAULT_TIMEOUT_VALUE (2 * 60 * 1000)
 
 typedef void (* DBusPendingFdsChangeFunction) (void *data);
 
-- 
2.43.0

