From 5ee56ca2928fab33623fed185705460a7f359b52 Mon Sep 17 00:00:00 2001
From: Izak Burger <isburger@gmail.com>
Date: Mon, 9 Jan 2023 15:34:14 +0200
Subject: [PATCH] evcharger: Update enumerations for status and mode

---
 ev_charger.py | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/ev_charger.py b/ev_charger.py
index f6408fd..ef31c3c 100644
--- a/ev_charger.py
+++ b/ev_charger.py
@@ -9,6 +9,7 @@ from register import *
 class EVC_MODE(IntEnum):
     MANUAL          = 0
     AUTO            = 1
+    SCHEDULED       = 2
 
 class EVC_CHARGE(IntEnum):
     DISABLED = 0
@@ -26,6 +27,13 @@ class EVC_STATUS(IntEnum):
     GND_ERROR       = 8
     WELD_CON        = 9
     CP_SHORTED      = 10
+    EARTH_LEAKAGE   = 11
+    UNDERVOLTAGE    = 12
+    OVERVOLTAGE     = 13
+    OVERTEMPERATURE = 14
+    STARTCHARGE     = 21
+    SWITCH_TO_3P    = 22
+    SWITCH_TO_1P    = 23
 
 class EVC_POSITION(IntEnum):
     OUTPUT = 0
-- 
2.7.4

