--- a/npm-build/lib/node_modules/node-red/node_modules/@node-red/editor-client/public/red/tours/welcome.js
+++ b/npm-build/lib/node_modules/node-red/node_modules/@node-red/editor-client/public/red/tours/welcome.js
@@ -14,19 +14,6 @@
                 "fr": "<p>Prenons un moment pour découvrir les nouvelles fonctionnalités de cette version.</p>"
             }
         },
-        {
-            title: {
-                "en-US": "Update notifications",
-                "ja": "更新の通知",
-                "fr": "Notifications de mise à jour"
-            },
-            image: 'images/update-notification.png',
-            description: {
-                "en-US": `<p>Stay up to date with notifications when there is a new Node-RED version available, or updates to the nodes you have installed</p>`,
-                "ja": `<p>新バージョンのNode-REDの提供や、インストールしたノードの更新があった時に、通知を受け取ることができます。</p>`,
-                "fr": `<p>Désormais vous recevrez une notification lorsqu'une nouvelle version de Node-RED ou une nouvelle version relative à un des noeuds que vous avez installés est disponible</p>`
-            }
-        },
         {
             title: {
                 "en-US": "Flow documentation",
