From d21c8327fb4705fd20b277a5d1aa37def2970e3c Mon Sep 17 00:00:00 2001
From: Ernst de Vries <edevries@victronenergy.com>
Date: Mon, 28 Dec 2015 15:40:11 +0100
Subject: [PATCH] Project file: add more generic install path

---
 dbus_modbustcp.pro | 17 ++++++++++++-----
 1 file changed, 12 insertions(+), 5 deletions(-)

diff --git a/dbus_modbustcp.pro b/dbus_modbustcp.pro
index 2bac645..981f754 100644
--- a/dbus_modbustcp.pro
+++ b/dbus_modbustcp.pro
@@ -2,11 +2,18 @@
 VERSION = 0.8.0
 REVISION = $$system(git --git-dir $$PWD/.git --work-tree $$PWD describe --always --dirty --tags)
 
-# Add more folders to ship with the application, here
-target.path = /opt/dbus_modbustcp
-csv.path = /opt/dbus_modbustcp
-csv.files = *.csv
-INSTALLS += target csv
+# Add more folders to ship with the application here
+unix {
+    bindir = $$(bindir)
+    DESTDIR = $$(DESTDIR)
+    isEmpty(bindir) {
+        bindir = /usr/local/bin
+    }
+    INSTALLS += target csv
+    target.path = $${DESTDIR}$${bindir}
+    csv.path = $${DESTDIR}$${bindir}
+    csv.files = *.csv
+}
 
 # Create a include file with VERSION / REVISION
 version_rule.target = $$OUT_PWD/version.h
-- 
1.9.1

