From 5b7ca49dc5777536adf7a6e18b2231af1277ff46 Mon Sep 17 00:00:00 2001
From: Jeroen Hofstee <jhofstee@victronenergy.com>
Date: Mon, 28 Dec 2015 16:46:29 +0100
Subject: [PATCH 2/2] s/dbus_modbustcp/dbus-modbustcp

---
 dbus_modbustcp.pro | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dbus_modbustcp.pro b/dbus_modbustcp.pro
index 041f936..c1c0e40 100644
--- a/dbus_modbustcp.pro
+++ b/dbus_modbustcp.pro
@@ -36,7 +36,7 @@ QT       -= gui
 QT       += network
 QT       += dbus
 
-TARGET = dbus_modbustcp
+TARGET = dbus-modbustcp
 CONFIG   += console
 CONFIG   -= app_bundle
 
-- 
1.9.1

