diff --git a/arch/arm/boot/dts/bbb-octo-venus.dts b/arch/arm/boot/dts/bbb-octo-venus.dts
new file mode 100644
index 00000000..cb2fca37
--- /dev/null
+++ b/arch/arm/boot/dts/bbb-octo-venus.dts
@@ -0,0 +1,33 @@
+/dts-v1/;
+
+#include "am33xx.dtsi"
+#include "am335x-bone-common.dtsi"
+#include "bb-ve-cape.dtsi"
+
+/ {
+	model = "TI AM335x BeagleBone Black with victronenergy cape";
+	compatible = "ti,am335x-bone-black", "ti,am33xx";
+};
+
+&ldo3_reg {
+	regulator-min-microvolt = <1800000>;
+	regulator-max-microvolt = <1800000>;
+	regulator-always-on;
+};
+
+&mmc1 {
+	vmmc-supply = <&vmmcsd_fixed>;
+};
+
+&mmc2 {
+	vmmc-supply = <&vmmcsd_fixed>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&emmc_pins>;
+	bus-width = <8>;
+	status = "okay";
+};
+
+&usb0 {
+	status = "okay";
+	dr_mode = "host";
+};
